src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs

```diff
@@ -539,10 +547,18 @@ private void AddLocalV3ResourceOptions(FindPackageByIdResource resource)
                     logger,
                     cancellationToken);
             }
-            catch (FatalProtocolException e) when (_ignoreFailedSources)
+            catch (FatalProtocolException e)
             {
-                await LogWarningAsync(id, e);
-                return null;
+                if (_ignoreFailedSources)
+                {
+                    await LogWarningAsync(id, e);
+                    return null;
+                }
+                else
+                {
+                    RestoreLogMessage logMessage = RestoreLogMessage.CreateError(e.LogCode, ExceptionUtilities.DisplayMessage(e));
```

Comment:
Same question as above about not terminating the operation completely. 

My concern around this is whether the experience here is consistent for the user. 

I'm curious what happens if you have the following. 
10 projects, with at least 1 unique package id each that needs to be sourced from a private feed.
How many prompts would the customer get if the wrong creds are provided? 

I think it'd be possible that we get an indeterministic amount of NU13** errors (we could see some per project). 
On the flip side, this might lead to successful restores for projects that won't need to download anything while before we would've just failed super early. I think the customers wouldn't mind that change :)

Reply:
working on fix. looks promising.